activerecord/test/cases/arel/visitors/postgres_test.rb

```diff
@@ -1,16 +1,32 @@
 # frozen_string_literal: true
 
+require "active_model/attribute"
+require "active_record"
+require "active_record/connection_adapters/postgresql_adapter"
 require_relative "../helper"
 
 module Arel
   module Visitors
     class PostgresTest < Arel::Spec
       before do
         @visitor = PostgreSQL.new Table.engine.connection
-        @table = Table.new(:users)
+        @table = Table.new(:users, type_caster: fake_pg_caster)
         @attr = @table[:id]
       end
 
+      # map that converts attribute names to a caster
+      def fake_pg_caster
```

Comment:
Why is this needed? It's not super obvious to me from the other changes.

Reply:
Without it, this error occurs:
```
NoMethodError: undefined method `type_for_attribute' for nil:NilClass
    arel/table.rb:111:in `type_for_attribute'
    arel/attributes/attribute.rb:13:in `type_caster'
    arel/visitors/postgresql.rb:84:in `visit_Arel_Nodes_HomogeneousIn'
```
My best guess is this table object is an incomplete stub. This pattern is also used in `arel/attributes/attribute_test.rb` and `arel/nodes/homogeneous_in_test.rb`

Reply:
It looks like `ActiveModel::Attribute::WithCastValue`'s `value_before_type_cast`, which is used in its `hash`, will contain `ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Array::Data` which doesn't have a consistent hash, which seems to be caused by the `PG::TextEncoder::Array` it contains.